src/DBmysql.php

```diff
@@ -1335,6 +1432,62 @@ public function buildInsert($table, $params)
         return $query;
     }
 
+    /**
+     * Builds an insert statement with parameter collection for prepared statements.
+     *
+     * This method returns SQL with ? placeholders and a QueryParameterBag
+     * containing the values to be bound. Use with executeWithParams().
+     *
+     * @since 11.0.0
+     *
+     * @param string $table  Table name
+     * @param QuerySubQuery|array  $params Array of field => value pairs or a QuerySubQuery for INSERT INTO ... SELECT
+     * @phpstan-param array<string, mixed>|QuerySubQuery $params
+     *
+     * @return array{sql: string, params: QueryParameterBag}
+     */
+    public function buildInsertWithParams($table, $params): array
```

Comment:
IMO this is overkill, the original buildInsert is enough you just need to pass the correct values as query params like in the POC.